compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -2116,7 +2116,7 @@ function getWriteErrorReason(abstractValue: AbstractValue): string {
   } else if (abstractValue.reason.has(ValueReason.ReactiveFunctionArgument)) {
     return "Mutating component props or hook arguments is not allowed. Consider using a local variable instead";
   } else if (abstractValue.reason.has(ValueReason.State)) {
-    return "Mutating a value returned from 'useState()', which should not be mutated. Use the setter function to update instead";
+    return "Mutating a value returned from 'useState() or useReducer()', which should not be mutated. Use the setter or dispatcher function to update instead";
```

Comment:
See above, let’s create a new Reducer variant and customize the error message for it.